tests/MongoDB.Driver.Core.Tests/Specifications/connection-string/TestRunner.cs

```diff
@@ -60,6 +58,172 @@ public void RunTestDefinition(JsonDrivenTestCase testCase)
+                                    {
+                                        var bsonDocument = new BsonDocument();
+                                        foreach (var innerTag in tag.Tags)
+                                        {
+                                            bsonDocument.Add(innerTag.Name, innerTag.Value);
+                                        }
+                                        bsonArray.Add(new BsonDocument(bsonDocument));
+                                    }
+                                }
+                                expectedOption.Value.Should().Be(bsonArray);
+                            }
+                            break;
+                        case "replicaset":
+                            connectionString.ReplicaSet.Should().Be(expectedOption.Value.AsString);
+                            break;
+                        case "retrywrites":
+                            AssertBoolean(connectionString.RetryWrites, expectedOption.Value);
+                            break;
+                        case "serverselectiontimeoutms":
+                            AssertTimeStamp(connectionString.ServerSelectionTimeout, expectedOption.Value);
+                            break;
+                        case "sockettimeoutms":
+                            AssertTimeStamp(connectionString.SocketTimeout, expectedOption.Value);
+                            break;
+                        case "ssl":
+#pragma warning disable 618
+                            AssertBoolean(connectionString.Ssl, expectedOption.Value);
+#pragma warning restore 618
+                            break;
+                        case "tls":
+                            AssertBoolean(connectionString.Tls, expectedOption.Value);
+                            break;
+                        case "tlsinsecure":
+                            AssertBoolean(connectionString.TlsInsecure, expectedOption.Value);
+                            break;
+                        case "w":
+                            {
+                                var w = WriteConcern.WValue.Parse(expectedOption.Value.ToString());
+                                connectionString.W.Should().Be(w);
+                            }
+                            break;
+                        case "wtimeoutms":
+                            AssertTimeStamp(connectionString.WTimeout, expectedOption.Value);
+                            break;
+                        case "zlibcompressionlevel":
+                            {
+                                var level = connectionString
+                                    .Compressors
+                                    .First(c => c.Type == CompressorType.Zlib)
+                                    .Properties
+                                    .Single();
+                                expectedOption.Value.Should().Be(level.Value.ToString());
+                            }
+                            break;
```

Comment:
Refactor to:

    case "zlibcompressionlevel":
        var zlibCompressionLevel = GetZlibCompresionLevel(connectionString);
        zlibCompressionLevel.Should().Be(expectedOption.Value.ToInt32());
        break;

with helper method:

    private int GetZlibCompresionLevel(ConnectionString connectionString)
    {
        return (int)connectionString
            .Compressors
            .First(c => c.Type == CompressorType.Zlib)
            .Properties
            .Single(kv => kv.Key == "Level")
            .Value;
    }